test/performance/range/gap_decorator_rand_write_benchmark.cpp

```diff
@@ -0,0 +1,150 @@
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <chrono>
+#include <cmath>
+#include <cstring>
+#include <random>
+#include <utility>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp>
+#include <seqan3/alignment/exception.hpp>
+#include <seqan3/alphabet/all.hpp>
+#include <seqan3/range/decorator/all.hpp>
+#include <seqan3/std/ranges>
+
+#include "gap_decorator_helper.hpp"
+
+using namespace seqan3;
+
+// Apply benchmarks with custom ranges for grid parameters sequence length and gap proportions
+static void CustomArguments(benchmark::internal::Benchmark* b) {
+    std::array<long long int, 5> gap_percentages = {1, 5, 25, 50, 75};
+    for (long long int seq_len = 4; seq_len <= (1 << 14); seq_len <<= 2)
+    {
+        for (auto gap_percentage : gap_percentages)
+            b->Args({seq_len, gap_percentage});
+    }
+}
+
+// ============================================================================
+//  insert at random position
+// ============================================================================
+template <typename gap_decorator_t, bool gapped_flag>
+static void insert_random(benchmark::State& state)
+{
+    unsigned int seq_len = state.range(0);
+    using size_type = typename gap_decorator_t::size_type;
+    using sequence_type = remove_cvref_t<detail::unaligned_seq_t<gap_decorator_t>>;
+    sequence_type seq(seq_len, 'A'_dna4);
+
+    // vector of sampled gap lengths for each position
+    std::vector<size_type> gaps(seq_len, 0);
+
+    // determine sum of gaps and non-gap symbols for not exceeding targeted sequence length
+    if constexpr(gapped_flag)
+    {
+        sample<size_type>(gaps, seq_len, state.range(1)/100.0);
+        resize<size_type, sequence_type>(gaps, seq, seq_len);
+    }
+
+    // initialize with (truncated) sequence and insert gaps from left to right
+    gap_decorator_t gap_decorator;
+    assign_unaligned(gap_decorator, seq);
+
+    // insert gaps before starting benchmark
+    if constexpr(gapped_flag)
+        insert_gaps<gap_decorator_t>(gaps, gap_decorator, seq_len);
+
+    std::mt19937 generator(time(0)); //Standard mersenne_twister_engine seeded with current time
+    std::uniform_real_distribution<> uni_dis{0.0, static_cast<double>(seq_len)};
+
+    // presample insert positions
+    std::vector<size_t> access_positions;
+    access_positions.resize(1 << 18);
+    for (size_t i = 0; i < access_positions.size(); ++i)
+        access_positions[i] = uni_dis(generator);
+    size_t j = 0;
+    for (auto _ : state)
+    {
+        state.PauseTiming();
+        auto it = std::ranges::begin(gap_decorator);
+        std::ranges::advance(it, access_positions[j++]);
```

Comment:
see above; do not advance every time, instead just increment the iterator every iteration which is constant time. also remove pausing of timer

Reply:
I don't get it :/. 
> do not advance every time, instead just increment the iterator every iteration

"every time" = "every iteration" or what am I missing here?

Reply:
Not sure, if I understand this remark, because this is a random write benchmark and not a sequential write one (e.g. insert from left to right), that's why for each measured operation the iterator is moved to a random position.

Reply:
Well, the problem is that advance always goes linearly from the beginning to the positions. This is very likely a lot slower than the insertion. We need to wait for the changes by @smehringer so that you can use the random_access iterator so you can jump in log-time instead of linear time at least.

Reply:
alternatively you can generate X number of iterators before the loop, instead of X number of access_positions. That's definitely better for the benchmark.

Reply:
Or just pause the timer, but this would still consume more CPU cycles then having iterators prepared. My new commit will implement your last proposal. We are getting close to the most environmental friendly solution ;)

Reply:
Wait, I cannot prepare iterators for the same gap_decorator instance. Because after a couple of insertions, the decorator might be moved in memory and hereby invalidating the pre-computed iterators (=> segfault).

Reply:
Yes, your are right. For the write benchmarks, you cannot use this approach. You should still use the random access iterator. @smehringer What is the state of #1139 ?

Reply:
And yes, if it cannot be avoided, stop the timer.

Reply:
The read benchmarks are currently testing the []-operator, so we are measuring the time of its iterator's private `jump` method. Pre-computing the iterators, means we measure `operator*` of the iterator, which should be constant for all approaches.